addons/im_livechat/static/src/legacy/public_livechat_chatbot.js

```diff
@@ -0,0 +1,874 @@
+                // user has already typed a message in -> trigger next step
+                this._chatbotSetIsTyping();
+                this.nextStepTimeout = setTimeout(
+                    this._chatbotTriggerNextStep.bind(this), this._chatbotMessageDelay);
+            } else {
+                this._chatbotEnableInput();
+            }
+        } else {
+            let triggerNextStep = true;
+            if (this._chatbotCurrentStep.chatbot_step_type === 'question_selection') {
+                const lastMessage = this._messages.length !== 0 ? this._messages[this._messages.length - 1] : null;
+                if (!lastMessage || lastMessage.getAuthorID() === this._livechat.getOperatorPID()[0]) {
+                    // if there is no last message or if the last message is from the bot
+                    // -> don't trigger the next step, we are waiting for the user to pick an option
+                    triggerNextStep = false;
+                }
+            }
+
+            if (triggerNextStep) {
+                let nextStepDelay = this._chatbotMessageDelay;
+                if (this._chatWindow.$('.o_livechat_chatbot_typing').length !== 0) {
+                    // special case where we already have a "is typing" message displayed
+                    // can happen when the previous step did not trigger any message posted from the bot
+                    // e.g: previous step was "forward_operator" and no-one is available
+                    // -> in that case, don't wait and trigger the next step immediately
+                    nextStepDelay = 0;
+                } else {
+                    this._chatbotSetIsTyping();
+                }
+
+                this.nextStepTimeout = setTimeout(
+                    this._chatbotTriggerNextStep.bind(this), nextStepDelay);
+            }
+        }
+
+        if (!this._chatbotIsDisplayRestartButton()) {
+            this._chatWindow.$('.o_livechat_chatbot_main_restart').addClass('d-none');
+        }
+     },
+     /**
+      * See '_chatbotSaveSession'.
+      *
+      * We retrieve the livechat uuid from the session cookie since the livechat Widget is not yet
+      * initialized when we restore the chatbot state.
+      *
+      * We also clear any older keys that store a previously saved chatbot session.
+      * (In that case we clear the actual browser's local storage, we don't use the localStorage
+      * object as it does not allow browsing existing keys, see 'local_storage.js'.)
+      *
+      * @private
+      */
+    _chatbotRestoreSession: function (sessionCookie) {
+        const browserLocalStorage = window.localStorage;
+        if (browserLocalStorage && browserLocalStorage.length) {
+            for (let i = 0; i < browserLocalStorage.length; i++) {
+                const key = browserLocalStorage.key(i);
+                if (key.startsWith('im_livechat.chatbot.state.uuid_') &&
```

Comment:
This part of the condition is not necessary
```
key.startsWith('im_livechat.chatbot.state.uuid_')
```

Reply:
... I don't understand :smile: 
This would remove all the localStorage keys that are not a chatbot session save, right?

I probably missed your point.

Reply:
no are are right, in my mind `key ===` and not `key !==` so indeed, it will remove all keys